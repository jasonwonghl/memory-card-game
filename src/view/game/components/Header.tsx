import React from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import { textVariants } from 'styles/styles'

interface Props {
  movesCount?: number
  onPress: () => void
}

const Header = ({ movesCount = 0, onPress }: Props) => {
  return (
    <View style={styles.container}>
      <TouchableOpacity testID={'RestartButton'} onPress={onPress}>
        <Text style={styles.restartButton}>Restart</Text>
      </TouchableOpacity>
      <View>
        <Text style={styles.movesLabel}>
          Steps:{' '}
          <Text testID={'StepsCountText'} style={styles.movesCount}>
            {movesCount}
          </Text>
        </Text>
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    height: 60,
    paddingVertical: 10,
    paddingHorizontal: 30,
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  restartButton: {
    color: textVariants.highlight,
    fontSize: textVariants.body.fontSize,
  },
  movesLabel: {
    fontSize: textVariants.subHeader.fontSize,
    color: textVariants.body.color,
    textTransform: 'uppercase',
  },
  movesCount: { color: textVariants.highlight },
})

export default Header
