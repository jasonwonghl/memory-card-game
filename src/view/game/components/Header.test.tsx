import { shallow, ShallowWrapper } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'
import Header from './Header'

describe('Header', () => {
  const mockOnPress = jest.fn()
  let wrapper: ShallowWrapper<any, any, React.Component<{}, {}, any>>
  let stepsCountText: ShallowWrapper<any, any, React.Component<{}, {}, any>>

  beforeEach(() => {
    wrapper = shallow(<Header movesCount={1} onPress={mockOnPress} />)

    stepsCountText = wrapper.findWhere(
      (node) => node.prop('testID') === 'StepsCountText'
    )
  })

  it('should render correctly', () => {
    const tree = renderer.create(
      <Header movesCount={1} onPress={mockOnPress} />
    )
    expect(tree).toMatchSnapshot()
  })

  it('should receive the moves count as props', () => {
    expect(stepsCountText.props().children).toEqual(1)
  })
})
