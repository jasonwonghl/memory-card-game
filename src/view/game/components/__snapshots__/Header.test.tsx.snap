// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "space-between",
      "paddingHorizontal": 30,
      "paddingVertical": 10,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="RestartButton"
  >
    <Text
      style={
        Object {
          "color": "dodgerblue",
          "fontSize": 18,
        }
      }
    >
      Restart
    </Text>
  </View>
  <View>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 28,
          "textTransform": "uppercase",
        }
      }
    >
      Steps:
       
      <Text
        style={
          Object {
            "color": "dodgerblue",
          }
        }
        testID="StepsCountText"
      >
        1
      </Text>
    </Text>
  </View>
</View>
`;
